#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_fix_bashism.dpatch by Francesco Namuri <francesco@namuri.it>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: this patch fixes bashism found in the scripts's program.

@DPATCH@
--- a/scripts/help.sh
+++ b/scripts/help.sh
@@ -9,13 +9,13 @@
 
 # detect user agents
 GNOME_BROWSER=`gconftool-2 --get /desktop/gnome/url-handlers/http/command | sed s/\"*%s\"*//`
-which konqueror &> /dev/null
+which konqueror > /dev/null 2>&1
 [ $? = 0 ] && KONQUEROR=konqueror
-which firefox &> /dev/null
+which firefox > /dev/null 2>&1
 [ $? = 0 ] && FIREFOX=firefox
-which mozilla &> /dev/null
+which mozilla > /dev/null 2>&1
 [ $? = 0 ] && MOZILLA=mozilla
-which yelp &> /dev/null
+which yelp > /dev/null 2>&1
 [ $? = 0 ] && YELP=yelp
 
 # convert index to kino for docbook
--- a/scripts/publish/bliptv_frame.sh
+++ b/scripts/publish/bliptv_frame.sh
@@ -35,9 +35,11 @@
 [ -z "$title" ] && title="Untitled"
 
 echo Login to blip.tv
-read -p "  Username: " username
+echo "Username: "
+read username
 stty -echo
-read -p "  Password: " password
+echo "Password: "
+read password
 echo
 stty echo
 
@@ -60,4 +62,4 @@
 	echo $result
 	echo 'Close the window when you are finished reading this error.'
 fi
-read
+read nop
--- a/scripts/publish/bliptv_project.sh
+++ b/scripts/publish/bliptv_project.sh
@@ -58,13 +58,15 @@
 ogg_file="${path}/${base}.ogg"
 
 echo Login to blip.tv
-read -p "  Username: " username
+echo "Username: "
+read username
 stty -echo
-read -p "  Password: " password
+echo "Password: "
+read password
 echo
 stty echo
-echo
-read -p "Do you want higher quality (slower and bigger to convert & upload)? [y/N] " quality
+echo "Do you want higher quality (slower and bigger to convert & upload)? [y/N] "
+read quality
 
 title=$(getXmlAttributeValue 'title' "$project_file")
 [ -z "$title" ] && title="Untitled"
@@ -166,4 +168,4 @@
 	echo $result
 	echo 'Close the window when you are finished reading this error.'
 fi
-read
+read nop
--- a/scripts/publish/echo.sh
+++ b/scripts/publish/echo.sh
@@ -1,5 +1,5 @@
 #!/bin/sh
 # A Kino publishing helper script that is used to display a message in a terminal window.
 echo $@
-read
+read nop
 
--- a/scripts/publish/frame.sh
+++ b/scripts/publish/frame.sh
@@ -7,9 +7,9 @@
 project_title="$4"
 
 if [ -n "$KINO_HOME" ] && [ -f "$KINO_HOME"/publish/frame.sh ]; then
-	. "$KINO_HOME"/publish/frame.sh "$frame_file" "$project_id" "$scene_title" "$project_title"
+	."$KINO_HOME"/publish/frame.sh "$frame_file" "$project_id" "$scene_title" "$project_title"
 elif [ -f "$HOME"/kino/publish/frame.sh ]; then
-	. "$HOME"/kino/publish/frame.sh "$frame_file" "$project_id" "$scene_title" "$project_title"
+	."$HOME"/kino/publish/frame.sh "$frame_file" "$project_id" "$scene_title" "$project_title"
 else
 	# detect terminal
 	GNOME_TERM=`gconftool-2 --get /desktop/gnome/applications/terminal/exec`
--- a/scripts/publish/project.sh
+++ b/scripts/publish/project.sh
@@ -15,9 +15,9 @@
 
 
 if [ -n "$KINO_HOME" ] && [ -f "$KINO_HOME"/publish/project.sh ]; then
-	. "$KINO_HOME"/publish/project.sh "$project_file" "$project_id" "$project_title"
+	."$KINO_HOME"/publish/project.sh "$project_file" "$project_id" "$project_title"
 elif [ -f "$HOME"/kino/publish/project.sh ]; then
-	. "$HOME"/kino/publish/project.sh "$project_file" "$project_id" "$project_title"
+	."$HOME"/kino/publish/project.sh "$project_file" "$project_id" "$project_title"
 elif [ -n "$KINO_PUBLISH_PROJECT" ]; then
 	"$KINO_PUBLISH_PROJECT" "$project_file" "$project_id" "$project_title"
 else
